java/daikon/dcomp/Premain.java

```diff
@@ -70,50 +70,13 @@
   protected static Set<String> problem_classes =
       new HashSet<>(
           Arrays.asList(
-              // Interfaces marked @FunctionalInterface should not be instrumented
-              // as they are used by LambdaExpressions.  Since BCEL won't let us
-              // inspect the annotations, we must manually add to problem classes.
-              "java.util.regex.Pattern$CharPredicate",
               // <clinit> gets a JNI error during initialization.
               "java.lang.StackTraceElement$HashedModules"));
 
   /** Set of methods known to cause problems when instrumented. */
   protected static Set<String> problem_methods =
       new HashSet<>(
           Arrays.asList(
-              // The following methods call sun.reflect.Reflection.getCallerClass().
-              // These methods are annotated with @CallerSensitive to
-              // indicate 'skip me when looking at call stack'.
-              // When we create the instrumented version of these methods we
-              // should include this annotation.  However, BCEL does not support
-              // this at this time.  So for now, we do not instrument these methods.
-              // Initially, we just had a couple methods listed, but it turns out
-              // that the junit tool references a large number of these methods.
-              "java.lang.Class.forName",
-              "java.lang.Class.newInstance",
-              "java.lang.Class.getClassLoader",
-              "java.lang.Class.getEnclosingMethod",
-              "java.lang.Class.getDeclaringClass",
-              "java.lang.Class.getEnclosingClass",
-              "java.lang.Class.getClasses",
-              "java.lang.Class.getField",
-              "java.lang.Class.getFields",
-              "java.lang.Class.getMethod",
-              "java.lang.Class.getMethods",
-              "java.lang.Class.getConstructor",
-              "java.lang.Class.getConstructors",
-              "java.lang.Class.getDeclaredClasses",
-              "java.lang.Class.getDeclaredField",
-              "java.lang.Class.getDeclaredFields",
-              "java.lang.Class.getDeclaredMethod",
-              "java.lang.Class.getDeclaredMethods",
-              "java.lang.Class.getDeclaredConstructor",
-              "java.lang.Class.getDeclaredConstructors",
-              "java.util.ResourceBundle.getBundle",
-              "java.util.ResourceBundle.clearCache",
-
-              // The below entries are temporary, until the bugs are fixed.
```

Comment:
Should this be retained?

Reply:
It turns out there appear to be no problem methods now - I deleted rest of list.

Reply:
Wow, that's great!